### PR TITLE
IT-1710: Support branches and add deploy

### PR DESCRIPTION
**Pull request by xschildw**
Do not merge before we confirm OK on branches etc. and verify the deployement environment (buckets, role, perms).
Missing invalidation after deploy.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,20 +1,51 @@
...
+  build-and-deploy:
+    name: build-and-deploy
+    # limit to this repository
+    if: github.repository == 'Sage-Bionetworks/SageAccountWeb'
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - name: SetupNode
+      # if in staging branch set then bucket name to staging bucket
+      - name: setup-ouputbucket-staging
```
Comment (jay-hodgson) [inline .github/workflows/build.yml]: missing "t", setup-ou**t**putbucket-staging

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,20 +1,51 @@
...
+        if: github.ref == 'refs/heads/staging'
+        run: echo 'BUCKET_NAME=staging.accounts.sagebionetworks.org' >> $GITHUB_ENV
+      # if in production branch then set bucket name to production bucket
+      - name: setup-ouputbucket-prod
```
Comment (jay-hodgson) [inline .github/workflows/build.yml]: here too.   setup-outputbucket-prod

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -1,20 +1,51 @@
...
+      - name: config-aws-creds
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AKID }}
+          aws-secret-access-key: ${{ secrets.AKSID }}
+          aws-region: us-east-1
+          mask-aws-account-id: true
+          role-to-assume: arn:aws:iam::797640923903:role/test-ghdeployer-GithubDeployerRole-V271Z804BS72
+          role-session-name: deploy-sageaccountweb-build
+          # TODO: fix role to allow this
```
Comment (jay-hodgson) [inline .github/workflows/build.yml:45]: Fix role to allow session tagging?

**Comment by jay-hodgson** [comment]
Looks like you are setting the bucket endpoint based on the github.ref, and running the deploy using aws credentials (assuming a role).

File: .github/workflows/build.yml (hunk 3 above)
Comment (xschildw) [reply .github/workflows/build.yml:45]: Yes, I'll fix in the deployer role then we remove this setting (and tag the session)

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**